src/frontend/src/components/ReactHookEditableList.tsx

```diff
@@ -1,35 +1,61 @@
-import { Grid, Button, IconButton, TextField } from '@mui/material';
-import DeleteIcon from '@mui/icons-material/Delete';
+import { Grid, Button, IconButton, TextField, InputLabel } from '@mui/material';
+import DeleteForeverOutlinedIcon from '@mui/icons-material/DeleteForeverOutlined';
 import { FieldArrayWithId, UseFieldArrayRemove, UseFormRegister, UseFieldArrayAppend } from 'react-hook-form';
+import { Box } from '@mui/system';
 
 interface ReactHookEditableListProps {
   name: string;
   ls: FieldArrayWithId[];
   register: UseFormRegister<any>;
   append: UseFieldArrayAppend<any, any>;
   remove: UseFieldArrayRemove;
+  bulletName?: string;
 }
 
-const ReactHookEditableList: React.FC<ReactHookEditableListProps> = ({ name, ls, register, append, remove }) => {
+const ReactHookEditableList: React.FC<ReactHookEditableListProps> = ({ name, ls, bulletName, register, append, remove }) => {
   return (
     <>
-      {ls.map((_element, i) => {
-        return (
-          <Grid item sx={{ display: 'flex', alignItems: 'center' }}>
-            <TextField required autoComplete="off" sx={{ width: 9 / 10 }} {...register(`${name}.${i}.detail`)} />
-            <IconButton type="button" onClick={() => remove(i)} sx={{ mx: 1, my: 0 }}>
-              <DeleteIcon />
-            </IconButton>
-          </Grid>
-        );
-      })}
+      <Grid container columnSpacing={{ xs: 4, sm: 2, md: 3 }}>
+        {ls.map((_element, i) => {
+          const formattedName = name
+            .split(/(?=[A-Z])/)
+            .map((name) => name.charAt(0).toUpperCase() + name.slice(1))
+            .join(' ');
+
+          return (
+            <Grid item xs={12} md={6} lg={4}>
```

Comment:
each of these grid items should have a unique key (can be made by doing {`${name}-${i}`}